{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "open_targets.ipynb",
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "metadata": {
        "id": "DQoBPVYywCKw"
      },
      "source": [
        "import requests\n",
        "import pandas as pd\n",
        "import json"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "3WzOWaRnwIAk"
      },
      "source": [
        "# query open targets and return json\n",
        "def gene_coloc_open_targets_query(geneid):\n",
        "  \"\"\"\n",
        "  queries open targets for colocalization data given gene id (ENS###)\n",
        "\n",
        "  input: geneid (ENS###)\n",
        "  output: coloc dataframe with the following columns:\n",
        "  phenotypeId\tqtlStudyId\th3\th4\tlog2h4h3\tstudy.studyId\tstudy.traitReported\t\n",
        "    study.pubAuthor\tleftVariant.id\ttissue.id\ttissue.name\n",
        "  \"\"\"\n",
        "  # query open targets for each gene in list\n",
        "  api_query = f'''query {{  \n",
        "    colocalisationsForGene(geneId: \"{geneid}\") {{\n",
        "      study{{\n",
        "        studyId\n",
        "        traitReported\n",
        "        pubAuthor\n",
        "      }}\n",
        "      leftVariant {{\n",
        "        id\n",
        "      }}\n",
        "      phenotypeId\n",
        "      tissue {{\n",
        "        id\n",
        "        name\n",
        "      }}\n",
        "      qtlStudyId\n",
        "      h3\n",
        "      h4\n",
        "      log2h4h3 \n",
        "    }}\n",
        "  }}'''\n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query': api_query})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{geneid} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def qtl_coloc_open_targets_query(variant_id):\n",
        "  \"\"\"\n",
        "  queries open targets for qtl data given variant_id in chr_pos_a1_a2 format \n",
        "    (hg38)\n",
        "\n",
        "  input: variant id in chr_pos_a1_a2 format (hg38)\n",
        "  output: dataframe with the following columns:\n",
        "  qtls gene.symbol gene.id\n",
        "  \"\"\"\n",
        "  api_query = f'''query {{\n",
        "    genesForVariant(variantId: \"{variant_id}\") {{\n",
        "      gene {{\n",
        "        id\n",
        "        symbol\n",
        "      }}\n",
        "      qtls{{\n",
        "        typeId\n",
        "        sourceId\n",
        "        aggregatedScore\n",
        "        tissues{{\n",
        "          tissue {{\n",
        "            id\n",
        "            name\n",
        "          }}\n",
        "          quantile\n",
        "          beta\n",
        "          pval\n",
        "        }}\n",
        "      }}\n",
        "    }}\n",
        "  }}'''\n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query': api_query})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{variant_id} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def phewas_per_variant_open_targets_query(variant_id):\n",
        "  \"\"\"\n",
        "  query open targets for phewas data\n",
        "\n",
        "  input: variant id (chr_pos_a1_a2 format with hg38 positions)\n",
        "  output: phewas dataframe with the following columns:\n",
        "  pval\tbeta\tnTotal\tnCases\toddsRatio\teaf\tse\tstudy.studyId\t\n",
        "    study.traitReported\tstudy.pubAuthor\n",
        "  \"\"\"\n",
        "  api_query = \"\"\"\n",
        "    query pheWAS($variantId: String!){\n",
        "      pheWAS(variantId: $variantId) {\n",
        "        associations {\n",
        "          study {\n",
        "            studyId\n",
        "            traitReported\n",
        "            pubAuthor\n",
        "          }\n",
        "          pval\n",
        "          beta\n",
        "          nTotal\n",
        "          nCases\n",
        "          oddsRatio\n",
        "          eaf\n",
        "          se\n",
        "        }\n",
        "      }\n",
        "    }\"\"\" \n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "  #set variables object\n",
        "  variables = {\"variantId\": variant_id}\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query':api_query, \"variables\":variables})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{variant_id} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def phewas_per_gene_open_targets_query(geneid):\n",
        "  api_query = \"\"\"\n",
        "    query studiesAndLeadVariantsForGene($geneId: String!){\n",
        "      studiesAndLeadVariantsForGene(geneId: $geneId) {\n",
        "        indexVariant {\n",
        "          id\n",
        "          rsId\n",
        "        }\n",
        "        study {\n",
        "          studyId\n",
        "          traitReported\n",
        "        }\n",
        "        pval\n",
        "        beta\n",
        "        oddsRatio\n",
        "        direction\n",
        "      }\n",
        "    }\"\"\"\n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "  #set variables object\n",
        "  variables = {\"geneId\": geneid}\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query':api_query, \"variables\":variables})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{geneid} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def search_rsid_open_targets_query(rsid):\n",
        "\n",
        "  \"\"\"\n",
        "  query open targets for snp ids in chr_pos_a1_a2 and allele freqs with rsid\n",
        "\n",
        "  input: rsid\n",
        "  output: ids and allele frequencies for different ancestry groups with columns:\n",
        "  id\trsId\tgnomadAFR\tgnomadAMR\tgnomadASJ\tgnomadEAS\tgnomadFIN\tgnomadNFE\t\n",
        "    gnomadNFEEST\tgnomadNFENWE\tgnomadNFESEU\tgnomadNFEONF\tgnomadOTH\n",
        "\n",
        "  Some information on the allele frequencies being pulled \n",
        "\n",
        "  \"gnomAD Allele frequency (African/African-American population)\"\n",
        "    gnomadAFR: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Latino/Admixed American population)\"\n",
        "    gnomadAMR: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Ashkenazi Jewish population)\"\n",
        "    gnomadASJ: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (East Asian population)\"\n",
        "    gnomadEAS: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Finnish population)\"\n",
        "    gnomadFIN: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish European population)\"\n",
        "    gnomadNFE: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean Estonian sub-population)\"\n",
        "    gnomadNFEEST: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean North-Western European sub-population)\"\n",
        "    gnomadNFENWE: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean Southern European sub-population)\"\n",
        "    gnomadNFESEU: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean Other non-Finnish European sub-population)\"\n",
        "    gnomadNFEONF: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Other (population not assigned) population)\"\n",
        "    gnomadOTH: Float\n",
        "    \"\"\"\n",
        "\n",
        "  api_query = \"\"\"\n",
        "  query search($queryString: String!){\n",
        "    search(queryString: $queryString) {\n",
        "      variants{\n",
        "        id\n",
        "        rsId\n",
        "        gnomadAFR\n",
        "        gnomadAMR\n",
        "        gnomadASJ\n",
        "        gnomadEAS\n",
        "        gnomadFIN\n",
        "        gnomadNFE\n",
        "        gnomadNFEEST\n",
        "        gnomadNFENWE\n",
        "        gnomadNFESEU\n",
        "        gnomadNFEONF\n",
        "        gnomadOTH\n",
        "      }\n",
        "    }\n",
        "  }\"\"\"\n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "  #set variables object\n",
        "  variables = {\"queryString\": rsid}\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query':api_query, \"variables\":variables})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{rsid} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def query_qtls(snp_list):\n",
        "  \"\"\"\n",
        "  uses qtl_coloc_open_targets_query() function to query open targets for qtls\n",
        "    given a list of snps in chr_pos_a1_a2 format (hg38)\n",
        "\n",
        "  input: list of snps (chr_pos_a1_a2 formatted with hg38 positions)\n",
        "  output: cleaned up dataframe of qtls with following columns:\n",
        "  snp\tgene_symbol\tgene_id\ttype\ttissue\tbeta\tpval\n",
        "  \"\"\"\n",
        "\n",
        "  total_qtls_df = pd.DataFrame()\n",
        "  for snp in snp_list:\n",
        "    qtl_query = qtl_coloc_open_targets_query(snp)\n",
        "    qtl_query_df = pd.json_normalize(qtl_query['data']['genesForVariant'])\n",
        "    if len(qtl_query_df) !=  0:\n",
        "      qtl_query_df['snp'] = snp\n",
        "      qtls_df = qtl_query_df.loc[qtl_query_df.qtls.map(lambda d: len(d)) > 0].reset_index(drop=True).copy()\n",
        "      total_qtls_df = total_qtls_df.append(qtls_df)\n",
        "  total_qtls_df = total_qtls_df.reset_index(drop=True)\n",
        "\n",
        "  final_qtls_df = pd.DataFrame()\n",
        "  for i, qtl in enumerate(total_qtls_df.qtls):\n",
        "    for j, qt in enumerate(qtl):\n",
        "      for k, tissue in enumerate(qt['tissues']):\n",
        "\n",
        "        qtl_dict = {\n",
        "            'snp': total_qtls_df.loc[i,'snp'],\n",
        "            'gene_symbol': total_qtls_df.loc[i,'gene.symbol'],\n",
        "            'gene_id': total_qtls_df.loc[i,'gene.id'],\n",
        "            'type': qt['typeId'],\n",
        "            'tissue': tissue['tissue']['name'],\n",
        "            'beta': tissue['beta'],\n",
        "            'pval': tissue['pval']\n",
        "        }\n",
        "        \n",
        "        qtl_dict_df = pd.DataFrame.from_records([qtl_dict])\n",
        "        final_qtls_df = final_qtls_df.append(qtl_dict_df).reset_index(drop=True)\n",
        "        \n",
        "  return final_qtls_df"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "SFm7D3mZbEAq",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "outputId": "3ff84133-5334-4c62-cd24-039cd5457140"
      },
      "source": [
        "# query coloc data given gene_id\n",
        "coloc_query = gene_coloc_open_targets_query('ENSG00000188906')\n",
        "coloc_query_df = pd.json_normalize(coloc_query['data']['colocalisationsForGene'])\n",
        "coloc_query_df.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>phenotypeId</th>\n",
              "      <th>qtlStudyId</th>\n",
              "      <th>h3</th>\n",
              "      <th>h4</th>\n",
              "      <th>log2h4h3</th>\n",
              "      <th>study.studyId</th>\n",
              "      <th>study.traitReported</th>\n",
              "      <th>study.pubAuthor</th>\n",
              "      <th>leftVariant.id</th>\n",
              "      <th>tissue.id</th>\n",
              "      <th>tissue.name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.183847</td>\n",
              "      <td>0.238134</td>\n",
              "      <td>0.373264</td>\n",
              "      <td>NEALE2_2395_1</td>\n",
              "      <td>Pattern 1 | hair/balding pattern</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_153901213_C_CA</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.222941</td>\n",
              "      <td>0.116175</td>\n",
              "      <td>-0.940364</td>\n",
              "      <td>NEALE2_1050</td>\n",
              "      <td>Time spend outdoors in summer</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_153987528_A_AAAAG</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.152196</td>\n",
              "      <td>0.422189</td>\n",
              "      <td>1.471959</td>\n",
              "      <td>NEALE2_30040_raw</td>\n",
              "      <td>Mean corpuscular volume</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_154078049_A_G</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.199173</td>\n",
              "      <td>0.239907</td>\n",
              "      <td>0.268452</td>\n",
              "      <td>NEALE2_30050_raw</td>\n",
              "      <td>Mean corpuscular haemoglobin</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_154095720_C_T</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.208134</td>\n",
              "      <td>0.210694</td>\n",
              "      <td>0.017637</td>\n",
              "      <td>NEALE2_864</td>\n",
              "      <td>Number of days/week walked 10+ minutes</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_154113167_A_T</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       phenotypeId  qtlStudyId  ...     tissue.id   tissue.name\n",
              "0  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "1  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "2  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "3  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "4  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "\n",
              "[5 rows x 11 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "PVO_iKlwwSdO",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "outputId": "e2446e26-426b-49d9-8f6e-27958c629690"
      },
      "source": [
        "# query qtls for a list of snps\n",
        "# query_qtls() uses qtl_coloc_open_targets_query() for each snp id in input list and returns combined and cleaned up dataframe\n",
        "test_snp_list = ['1_154453788_C_T', '1_154445939_T_C']\n",
        "qtls_df = query_qtls(test_snp_list)\n",
        "qtls_df.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>snp</th>\n",
              "      <th>gene_symbol</th>\n",
              "      <th>gene_id</th>\n",
              "      <th>type</th>\n",
              "      <th>tissue</th>\n",
              "      <th>beta</th>\n",
              "      <th>pval</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>UBE2Q1</td>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (eQTLGen)</td>\n",
              "      <td>-0.047601</td>\n",
              "      <td>8.819400e-09</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (GTEX v7)</td>\n",
              "      <td>-0.106938</td>\n",
              "      <td>3.641230e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Macrophage listeria (NEDELEC 2016)</td>\n",
              "      <td>0.153303</td>\n",
              "      <td>1.443160e-06</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Monocyte r848 (QUACH 2016)</td>\n",
              "      <td>0.119603</td>\n",
              "      <td>2.832830e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Colon transverse (GTEX v7)</td>\n",
              "      <td>-0.269436</td>\n",
              "      <td>7.835600e-06</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "               snp gene_symbol  ...      beta          pval\n",
              "0  1_154453788_C_T      UBE2Q1  ... -0.047601  8.819400e-09\n",
              "1  1_154453788_C_T        IL6R  ... -0.106938  3.641230e-07\n",
              "2  1_154453788_C_T        IL6R  ...  0.153303  1.443160e-06\n",
              "3  1_154453788_C_T        IL6R  ...  0.119603  2.832830e-07\n",
              "4  1_154453788_C_T        IL6R  ... -0.269436  7.835600e-06\n",
              "\n",
              "[5 rows x 7 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 7
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "lrnLMv-7H7G7",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "outputId": "b5052dcd-4d41-496e-c2ae-44355a6b02bf"
      },
      "source": [
        "# query phewas data for a given snp\n",
        "phewas_variant_query = phewas_per_variant_open_targets_query('1_154453788_C_T')\n",
        "phewas_variant_query_df = pd.json_normalize(phewas_variant_query['data']['pheWAS']['associations'])\n",
        "phewas_variant_query_df.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pval</th>\n",
              "      <th>beta</th>\n",
              "      <th>nTotal</th>\n",
              "      <th>nCases</th>\n",
              "      <th>oddsRatio</th>\n",
              "      <th>eaf</th>\n",
              "      <th>se</th>\n",
              "      <th>study.studyId</th>\n",
              "      <th>study.traitReported</th>\n",
              "      <th>study.pubAuthor</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>0.032147</td>\n",
              "      <td>0.043068</td>\n",
              "      <td>361141</td>\n",
              "      <td>5182.0</td>\n",
              "      <td>1.044009</td>\n",
              "      <td>0.409865</td>\n",
              "      <td>0.020101</td>\n",
              "      <td>NEALE2_20002_1398</td>\n",
              "      <td>Pneumonia | non-cancer illness code, self-repo...</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>0.006910</td>\n",
              "      <td>0.004799</td>\n",
              "      <td>350404</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.409840</td>\n",
              "      <td>0.001776</td>\n",
              "      <td>NEALE2_1289</td>\n",
              "      <td>Cooked vegetable intake</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>0.024036</td>\n",
              "      <td>0.018557</td>\n",
              "      <td>117763</td>\n",
              "      <td>53389.0</td>\n",
              "      <td>1.018730</td>\n",
              "      <td>0.408367</td>\n",
              "      <td>0.008223</td>\n",
              "      <td>NEALE2_20446</td>\n",
              "      <td>Ever had prolonged feelings of sadness or depr...</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>0.000109</td>\n",
              "      <td>0.028852</td>\n",
              "      <td>361141</td>\n",
              "      <td>41934.0</td>\n",
              "      <td>1.029273</td>\n",
              "      <td>0.409865</td>\n",
              "      <td>0.007457</td>\n",
              "      <td>NEALE2_20002_1111</td>\n",
              "      <td>Asthma | non-cancer illness code, self-reported</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>0.009270</td>\n",
              "      <td>0.259938</td>\n",
              "      <td>361141</td>\n",
              "      <td>207.0</td>\n",
              "      <td>1.296850</td>\n",
              "      <td>0.409865</td>\n",
              "      <td>0.099904</td>\n",
              "      <td>NEALE2_20003_1141180392</td>\n",
              "      <td>Cefalexin | treatment/medication code</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       pval  ...  study.pubAuthor\n",
              "0  0.032147  ...     UKB Neale v2\n",
              "1  0.006910  ...     UKB Neale v2\n",
              "2  0.024036  ...     UKB Neale v2\n",
              "3  0.000109  ...     UKB Neale v2\n",
              "4  0.009270  ...     UKB Neale v2\n",
              "\n",
              "[5 rows x 10 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "id": "I9c_mhd5xG4a",
        "outputId": "7a21f784-f7fa-4837-c078-1f94baf7d5bf"
      },
      "source": [
        "# query phewas data for a given gene\n",
        "phewas_gene_query = phewas_per_gene_open_targets_query('ENSG00000188906')\n",
        "phewas_gene_query_df = pd.json_normalize(phewas_gene_query['data']['studiesAndLeadVariantsForGene'])\n",
        "phewas_gene_query_df.head()"
      ],
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pval</th>\n",
              "      <th>beta</th>\n",
              "      <th>oddsRatio</th>\n",
              "      <th>direction</th>\n",
              "      <th>indexVariant.id</th>\n",
              "      <th>indexVariant.rsId</th>\n",
              "      <th>study.studyId</th>\n",
              "      <th>study.traitReported</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>2.000000e-10</td>\n",
              "      <td>0.032</td>\n",
              "      <td>NaN</td>\n",
              "      <td>+</td>\n",
              "      <td>12_40212861_T_C</td>\n",
              "      <td>rs1907631</td>\n",
              "      <td>GCST007131</td>\n",
              "      <td>Low density lipoprotein cholesterol levels</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>6.000000e-15</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>None</td>\n",
              "      <td>12_40227006_C_G</td>\n",
              "      <td>rs1491942</td>\n",
              "      <td>GCST001445_2</td>\n",
              "      <td>Parkinson's disease [EA]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>9.000000e-09</td>\n",
              "      <td>0.030</td>\n",
              "      <td>NaN</td>\n",
              "      <td>+</td>\n",
              "      <td>12_40212861_T_C</td>\n",
              "      <td>rs1907631</td>\n",
              "      <td>GCST007134</td>\n",
              "      <td>Total cholesterol levels</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>5.000000e-14</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1.1550</td>\n",
              "      <td>+</td>\n",
              "      <td>12_40220632_C_T</td>\n",
              "      <td>rs76904798</td>\n",
              "      <td>GCST002544</td>\n",
              "      <td>Parkinson's disease</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>4.236247e-08</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.7593</td>\n",
              "      <td>-</td>\n",
              "      <td>12_40320530_T_C</td>\n",
              "      <td>rs11564177</td>\n",
              "      <td>GCST005527</td>\n",
              "      <td>Psoriasis</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "           pval  ...                         study.traitReported\n",
              "0  2.000000e-10  ...  Low density lipoprotein cholesterol levels\n",
              "1  6.000000e-15  ...                    Parkinson's disease [EA]\n",
              "2  9.000000e-09  ...                    Total cholesterol levels\n",
              "3  5.000000e-14  ...                         Parkinson's disease\n",
              "4  4.236247e-08  ...                                   Psoriasis\n",
              "\n",
              "[5 rows x 8 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "C0dvAkM_CLV5",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 80
        },
        "outputId": "2a4ba5af-9774-4904-a1af-b4056a267f49"
      },
      "source": [
        "# get hg38 chr_pos_a1_a2 and allele freqs for given rsid\n",
        "rsid_query = search_rsid_open_targets_query(\"rs4129267\")\n",
        "rsid_query_df = pd.json_normalize(rsid_query['data']['search']['variants'])\n",
        "rsid_query_df.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>id</th>\n",
              "      <th>rsId</th>\n",
              "      <th>gnomadAFR</th>\n",
              "      <th>gnomadAMR</th>\n",
              "      <th>gnomadASJ</th>\n",
              "      <th>gnomadEAS</th>\n",
              "      <th>gnomadFIN</th>\n",
              "      <th>gnomadNFE</th>\n",
              "      <th>gnomadNFEEST</th>\n",
              "      <th>gnomadNFENWE</th>\n",
              "      <th>gnomadNFESEU</th>\n",
              "      <th>gnomadNFEONF</th>\n",
              "      <th>gnomadOTH</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>rs4129267</td>\n",
              "      <td>0.137483</td>\n",
              "      <td>0.509434</td>\n",
              "      <td>0.451724</td>\n",
              "      <td>0.379534</td>\n",
              "      <td>0.301957</td>\n",
              "      <td>0.375928</td>\n",
              "      <td>0.340299</td>\n",
              "      <td>0.389706</td>\n",
              "      <td>0.528302</td>\n",
              "      <td>0.388941</td>\n",
              "      <td>0.331801</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                id       rsId  gnomadAFR  ...  gnomadNFESEU  gnomadNFEONF  gnomadOTH\n",
              "0  1_154453788_C_T  rs4129267   0.137483  ...      0.528302      0.388941   0.331801\n",
              "\n",
              "[1 rows x 13 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 6
        }
      ]
    }
  ]
}